.github/workflows/index-file-validation.yml

```diff
@@ -0,0 +1,114 @@
+name: "Validate index.json files"
+
+on:
+  pull_request:
+
+jobs:
+  validate-json:
+    name: "📋 Validate changed JSON files"
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+
+    steps:
+      - name: "☁️ Checkout repository"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: "🔎 Detect relevant file changes"
+        id: filter
+        uses: ./.github/actions/detect-file-changes
+        with:
+          file-patterns: |
+            - 'metadata/index.json'
+            - 'metadata/*/*/index.json'
+            - 'tests/src/index.json'
+            - 'tests/src/*/*/*/index.json'
+
+      - name: "🔧 Setup Java"
+        if: steps.filter.outputs.changed == 'true'
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'graalvm'
+          java-version: '21'
+
+      - name: "🕸️ Populate matrix"
+        id: set-matrix
+        if: steps.filter.outputs.changed == 'true'
+        run: |
+          ./gradlew generateChangedIndexFileMatrix -PbaseCommit=${{ github.event.pull_request.base.sha }} -PnewCommit=${{ github.event.pull_request.head.sha }}
```

Comment:
Why do we need a matrix for this? It takes 1 s to do, so we can do it all at once.